### PR TITLE
[release/6.0] Fixing the handling of Positive NaN in Math.Min for float/double

### PR DESCRIPTION
**Pull request by tannergooding**
Backport of #70795

/cc @tannergooding 

## Customer Impact

A customer reported this issue as https://github.com/dotnet/runtime/issues/70775

Math.Max and Math.Min are expected to propagate `NaN` back to the caller if either input is `NaN`. In .NET 6, this was regressed such that `Math.Min` would not propagate a positive `NaN` back to the caller which breaks this expectation.

This was not caught because `double.NaN`, the `NaN` returned by most hardware, and the `NaN` that the C# compiler and JIT compiler normalizes to is a negative `NaN`.

`Math.Max` was working as expected but tests explicitly validating its behavior in the face of positive NaN are included.

## Testing

To help ensure the behavior of the `Max`, `MaxMagnitude`, `Min`, and `MinMagnitude` tests validating both `positive` and `negative` `NaN` were added. This is achieved by explicitly toggling the sign-bit via a bitwise manipulation to help ensure that JIT optimizations (current or future) don't kick in.

The customer that reported the issue in https://github.com/dotnet/runtime/issues/70775 was asked to validate the fix as well and was able to confirm that it does indeed fix their issue.

## Risk

`Min` will take a small perf hit bringing it inline with `Max` (previously it was slightly faster).

It is not expected that anyone would be depending on the accidental break, but its possible that someone raises an issue.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Math.cs [hunk 1]
```diff
@@ -978,12 +978,17 @@ public static double Min(double val1, double val2)
             // This matches the IEEE 754:2019 `minimum` function
             //
             // It propagates NaN inputs back to the caller and
-            // otherwise returns the larger of the inputs. It
-            // treats +0 as larger than -0 as per the specification.
+            // otherwise returns the lesser of the inputs. It
+            // treats +0 as lesser than -0 as per the specification.
```
Comment (jgiannuzzi) [inline src/libraries/System.Private.CoreLib/src/System/Math.cs:982]: I guess that this comment should be inverted like this for the `Min` variants:

```suggestion
            // treats -0 as lesser than +0 as per the specification.
```

Sorry for not spotting this earlier on the original PR!

**Review by jgiannuzzi — COMMENTED** [review]
Thank you so much for backporting this fix @tannergooding!

File: src/libraries/System.Private.CoreLib/src/System/Math.cs [hunk 2]
```diff
@@ -1030,12 +1035,17 @@ public static float Min(float val1, float val2)
...
```
Comment (tannergooding) [inline src/libraries/System.Private.CoreLib/src/System/Math.cs:1039]:
```suggestion
            // treats -0 as lesser than +0 as per the specification.
```

File: src/libraries/System.Private.CoreLib/src/System/MathF.cs [hunk 3]
```diff
@@ -246,8 +246,8 @@ public static float MinMagnitude(float x, float y)
             // This matches the IEEE 754:2019 `minimumMagnitude` function
             //
             // It propagates NaN inputs back to the caller and
-            // otherwise returns the input with a larger magnitude.
-            // It treats +0 as larger than -0 as per the specification.
+            // otherwise returns the input with a lesser magnitude.
+            // It treats +0 as lesser than -0 as per the specification.
```
Comment (tannergooding) [inline src/libraries/System.Private.CoreLib/src/System/MathF.cs:250]:
```suggestion
            // It treats -0 as lesser than +0 as per the specification.
```

File: src/libraries/System.Private.CoreLib/src/System/Math.cs [hunk 4]
```diff
@@ -1078,8 +1088,8 @@ public static double MinMagnitude(double x, double y)
             // This matches the IEEE 754:2019 `minimumMagnitude` function
             //
             // It propagates NaN inputs back to the caller and
-            // otherwise returns the input with a larger magnitude.
-            // It treats +0 as larger than -0 as per the specification.
+            // otherwise returns the input with a lesser magnitude.
+            // It treats +0 as lesser than -0 as per the specification.
```
Comment (tannergooding) [inline src/libraries/System.Private.CoreLib/src/System/Math.cs:1092]:
```suggestion
            // It treats -0 as lesser than +0 as per the specification.
```

**Review by jeffhandley — APPROVED** [review]

**Comment by nojaf** [comment]
Many thanks @tannergooding and @jeffhandley!
Wonderful to see this fix and backport in such a short time!

**Comment by tannergooding** [comment]
Approved over e-mail.

**Review by stephentoub — APPROVED** [review]

**Pull request merged**